OADP-4001 Self-Service (#89043)
openshift/openshift-docs: modules/oadp-self-service-overview.adoc

** You keep administrator control over non-administrator operations through `DataProtectionApplication` configuration and policies.

* As a namespace admin user:
** You can create backup and restore custom resources for your authorized namespace.
** You can create dedicated backup storage locations in your authorized namespace.
** You have secure access to backup logs and status information.

[id="oadp-self-service-overview-namespace-scope_{context}"]
= What namespace-scoped backup and restore means

{oadp-short} Self-Service ensures that namespace admin users can only operate within their authorized namespace. For example, if you do not have access to a namespace, as a namespace admin user, you cannot back up that namespace.

A namespace admin user cannot access backup and restore data of other users.

The cluster administrator enforces the access control through custom resources (CRs) that securely manage the backup and restore operations.

Additionally, the cluster administrator can control the allowed options within the CRs, restricting certain operations for added security by using `spec` enforcements in the `DataProtectionApplication` (DPA) CR.

> Namespace `admin` users can perform the following Self-Service operations:

Author: Done.

// Module included in the following assemblies:
//
// backup_and_restore/application_backup_and_restore/oadp-self-service/oadp-self-service.adoc

:_mod-docs-content-type: CONCEPT
[id="oadp-self-service-overview_{context}"]
= About {oadp-short} Self-Service

> In the earlier versions of {oadp-full}, you needed the `cluster-admin` role to perform {oadp-short} operations such as backing up and restoring an application, creating a backup storage location, and so on. For more details on cluster roles, see "Default cluster roles".

Author: removed this.

> With {oadp-short} 1.5.0 onward, you do not need cluster-wide administrator privileges to perform the backup and restore operations. 

Reviewer: ```suggestion
From {oadp-short} 1.5.0 onward, you do not need cluster-wide administrator privileges to perform the backup and restore operations. 
```

> The {oadp-short} Self-Service feature provides secure self-service data protection capabilities for users without administrator privileges, while maintaining proper access controls.

Reviewer: ```suggestion
The {oadp-short} Self-Service feature provides secure self-service data protection capabilities for users without administrator privileges while maintaining proper access controls.
```

> As a non-admin user, you can back up, and restore applications deployed in your authorized namespace on the cluster. This means, you need to have appropriate access to your namespace.

Reviewer: ```suggestion
As a non-admin user, you can back up and restore applications deployed in your authorized namespace on the cluster. This means, you need to have appropriate access to your namespace.
```

The {oadp-short} cluster administrator creates a non-admin user and provides the necessary access to the non-admin user to perform {oadp-short} Self-Service actions.

Note that, {oadp-short} Self-service does not support cross-cluster restore.

{oadp-short} Self-Service offers the following benefits:

* As a cluster administrator:
** You allow namespace-scoped backup and restore operations to a non-admin user. This means, a non-admin user cannot access a namespace that they are not authorized to.
** You keep administrator control over non-administrator operations through templates and policies.

* As a non-administrator:
** You can create dedicated backup storage locations in your authorized namespace.

> ** You have secure access to backup logs and status information.

Author: Added this line.

> ** You have secure access to backup logs and status information.

Reviewer: ```suggestion
** You have secure access to back up logs and status information.
```

Author: Hey Andy, isn't the noun form - `backup logs` correct in this case? It means, a non-admin user has access to logs of the backup process.

= What namespace-scoped backup and restore means

> The {oadp-short} Self-Service feature ensures that non-admin users can only operate within their authorized namespace. For example, if you do not have access to a namespace, as a non-admin user, you cannot back up that namespace.

Reviewer: ```suggestion
{oadp-short} Self-Service ensures that non-admin users can only operate within their authorized namespace. For example, if you do not have access to a namespace, as a non-admin user, you cannot back up that namespace.
```

// Module included in the following assemblies:
//
// backup_and_restore/application_backup_and_restore/oadp-self-service/oadp-self-service.adoc

:_mod-docs-content-type: CONCEPT
[id="oadp-self-service-overview_{context}"]
= {oadp-short} Self-Service overview

> In the earlier versions of {oadp-full}, you needed the `cluster-admin` role to perform {oadp-short} operations such as backing up and restoring an application, creating a backup storage location, and so on. For more details on cluster roles, see "Default cluster roles ".

Author: Linking is bit tricky in the docs repo. We are not allowed to use hard links in a module. We are told to add links in the "Additional resources" section at the bottom. I know it is totally not useful. I will work with Andy to see if can find another solution to create a hyperlink here. Thanks.

> From {oadp-short} 1.5.0 onward, you do not need the `cluster-admin` role to perform the backup and restore operations. You can use {oadp-short} with the `admin` role. The `admin` role has administrator access only to the namespace the user is assigned to.

Author: I completely agree to the confusion created by `non-admin` user. I have asked a question a slack thread. And depending on the response, I will take action on this comment and the rest of the similiar comments. Thanks.

> From {oadp-short} 1.5.0 onward, you do not need the `cluster-admin` role to perform the backup and restore operations. You can use {oadp-short} with the `admin` role. The `admin` role has administrator access only to the namespace the user is assigned to.

Author: Using the term `namespace admin` everywhere in the docs.

> From {oadp-short} 1.5.0 onward, you do not need the `cluster-admin` role to perform the backup and restore operations. You can use {oadp-short} with the namespace `admin` role. The namespace `admin` role has administrator access only to the namespace the user is assigned to.

Author: done.

The {oadp-short} Self-Service feature provides secure self-service data protection capabilities for users without `cluster-admin` privileges while maintaining proper access controls.

The {oadp-short} cluster administrator creates a user with the `admin` role and provides the necessary Role Based Access Controls (RBAC) to the user to perform {oadp-short} Self-Service actions. As this user has limited access compared to the `cluster-admin` role, this user is referred to as a non-admin user. 

As a non-admin user, you can back up and restore applications deployed in your authorized namespace on the cluster.

> Note that, {oadp-short} Self-service does not support cross-cluster restore.

Author: Removed this line from here and added to the "{oadp-short} Self-Service unsupported features" section.

* As a cluster administrator:
** You allow namespace-scoped backup and restore operations to a namespace admin user. This means, a namespace admin user cannot access a namespace that they are not authorized to.

> ** You keep administrator control over non-administrator operations through templates and policies.

Reviewer 2: What do we mean here when we say templates ?

Author: re-worded for clarity. thanks.

* As a namespace admin user:

> ** You can create backup and restore custom resources.

Author: We are introducing the terms NAB and NAR a little later in the "OADP Self-Service custom resources" section. So I have re-worded the sentence for more clarity. Thanks.

* As a non-administrator:
** You can create backup and restore custom resources.
** You can create dedicated backup storage locations in your authorized namespace.
** You have secure access to backup logs and status information.

= What namespace-scoped backup and restore means

{oadp-short} Self-Service ensures that non-admin users can only operate within their authorized namespace. For example, if you do not have access to a namespace, as a non-admin user, you cannot back up that namespace.

This also means that a non-admin user cannot access backup and restore data of other users.

> The cluster administrator enforces the access control through custom resources that securely manage the backup and restore operations.

Author: added this info. thanks.

Non-admin users can:

* Create and manage backups of their authorized namespaces.
* Restore data to their authorized namespaces.
* Configure their own backup storage locations.
* Check backup and restore status.

> * Access relevant logs.

Author: corrected this line.

> * Create and manage backups of their authorized namespaces.

Reviewer 2: @mpryc A qq as per current design we only allow backing up a single namespace at a time, right ?  Do you think we should mentioned that here.
